### DISCUSSION TITLE
TypeError: loop of ufunc does not support argument 0 of type float which has no callable sqrt method

### DISCUSSION THREAD

**Discussion by LovisaBostrom** [pastas/pastas#297] [Q&A] [answered]
Hi,

My previous issue in discussion #296 was resolved after updating Pastas. However, when running the model after the update we encountered a new problem with the following error message:

TypeError: loop of ufunc does not support argument 0 of type float which has no callable sqrt method

I am therefore wondering if the update has made the simulation more sensitive to the quality of data?

What we are trying to do is perform several simulations with different stresses for example only with recharge and then adding leakage and/or infiltration as well models (as in the code I pasted below)

We can resolve this issue by choosing the solver LeastSquares instead of Lmfitsolve! However, since you state on the webpage that the Lmfitsolve is recommended when using a Wellmodel and some of our parameter values are much smaller than 1 we are hesitant in using this approach. Do you have any thoughts?

Thanks in advance!

```
# "settings"
ob_func=ps.LmfitSolve
rfun=ps.Gamma #responsefunction
NO=False #Noise
repo=False #report
rch=ps.rch.Linear()
date_TS='2019-07-01'
date_max='2021-04-01'
date_min= '2018-10-01'

# Model with only recharge
ml_WOL=ps.Model(Ho, name=WELL)
sm = ps.RechargeModel(rain, evap, rfunc=rfun, recharge=rch, name="rch")
ml_WOL.add_stressmodel(sm)
ml_WOL.solve( tmin=date_min, report=repo, solver=ob_func, noise=NO)

#Model with recharge + leakage
m2=ps.Model(Ho, name=WELL)
sm2 = ps.RechargeModel(rain, evap, rfunc=rfun, recharge=rch, name="rch")
m2.add_stressmodel(sm2)
ws2 = ps.WellModel(stresses2, rfunc=ps.HantushWellModel, name="leakage", distances=distances2, settings="well")
m2.add_stressmodel(ws2)
m2.solve( tmin=date_min,report=repo, solver=ob_func, noise=NO)
```

**Comment by raoulcollenteur** [comment, marked as answer]
This is related to https://github.com/pastas/pastas/issues/295 and was solved in the Dev-branch, so the first option is switching to that. Not sure when we'll update the Master-branch and Pypi to fix it. Maybe it's actually good to do an intermediate update.

However, if no covariance matrix is returned by Lmfit it is likely that something else is wrong with the model. What could possibly help is to rescale the stresses2, such that you're not estimating very small parameter values, and then just stick to ps.LeastSquares to solve the model.

Let me know if this helps!
Cheers, Raoul

**Reply by LovisaBostrom** [reply]
Yes switched to Dev-branch and that fixed it!

Thank you very much

**Reply by raoulcollenteur** [reply]
Cool! Sorry for the inconvenience! I'll let you know when we upgrade the stable version to fix this problem :)

**Comment by nikhilsaharan33** [comment]
You need to make sure you are using float/int values. The datatype must not be anything else.
